stack/building-with-pip.rst

```diff
@@ -0,0 +1,274 @@
+Middleware packages use a dual license for distribution, with the `PyPI`_ package declaring the BSD 3-clause license.
+Most Science Pipelines packages use the GPLv3 license and for those packages the ``pyproject.toml`` should use:
+
+.. code-block:: toml
+
+    license = {text = "GPLv3+ License"}
+    classifiers = [
+      "License :: OSI Approved :: GNU General Public License v3 or later (GPLv3+)",
+    ]
+
+Every Rubin DM Science Pipelines package should be owned by the special DM account and that should always be included in the ``authors`` section.
+Additional authors can be included if required.
+
+The ``dependencies`` section can only refer to packages that are available on `PyPI`_ since this is the section that will be read during ``pip install``.
+It should include all Python packages that would normally be included as part of the ``rubin-env`` Conda environment.
+Do not include optional packages in this list or packages that are only needed to run the test code.
+
+This can be achieved with something like the following:
+
+.. code-block:: toml
+
+  [project.optional-dependencies]
+  test = [
+      "pytest >= 3.2",
+      "numpy >= 1.17",
+      "matplotlib >= 3.0.3",
+      "pyarrow >= 0.16",
+      "pandas >= 1.0",
+  ]
+
+and would be installed automatically if ``pip`` is called with:
+
+.. code-block:: bash
+
+  $ pip install lsst-yourpackage[test]
+
+
+Setuptools Configuration
+------------------------
+
+For ``setuptools`` builds, the usual default build system for our packages, additional configuration is needed so that the python files and data files can be located.
+For example, in ``daf_butler`` there is this configuration:
+
+.. code-block:: toml
+
+    [tool.setuptools.packages.find]
+    where = ["python"]
+
+    [tool.setuptools]
+    zip-safe = true
+    license-files = ["COPYRIGHT", "LICENSE", "bsd_license.txt", "gpl-v3.0.txt"]
+
+    [tool.setuptools.package-data]
+    "lsst.daf.butler" = ["py.typed", "configs/*.yaml", "configs/*/*.yaml"]
+
+    [tool.setuptools.dynamic]
+    version = { attr = "lsst_versions.get_lsst_version" }
+
+This tells ``setuptools`` that the python files are in a ``python/`` directory and what additional non-python files should be included in the distribution.
+
+The license-files section should reflect the specific needs of your package.
+
+When making a `PyPI`_ distribution, the package should work without relying on the EUPS ``$PACKAGE_DIR`` variable being set.
+This means that any supplementary data such as those that would go in a ``config/`` or ``policy/`` directory should instead be included inside the ``python/`` directory and be accessed using the standard package resources APIs (such as `importlib.resources` or ``lsst.resources.ResourcePath``).
+These files must then be listed explicitly in the ``package-data`` section of the configuration file, as can be shown in the above example (the ``py.typed`` file tells Mypy that type annotations are defined for this package).
```

Comment:
Now that I understand that `package-data` includes _all non-python files_, `py.typed` doesn't need to be called out specially.